// @aws-sdk/client-timestream-write/dist-es/schemas/schemas_0.js
const _A = "Arn";
const _ADE = "AccessDeniedException";
const _Ad = "Address";
const _BLPR = "BatchLoadProgressReport";
const _BLT = "BatchLoadTask";
const _BLTD = "BatchLoadTaskDescription";
const _BLTL = "BatchLoadTaskList";
const _BLTa = "BatchLoadTasks";
const _BM = "BytesMetered";
const _BN = "BucketName";
const _CA = "CommonAttributes";
const _CBLT = "CreateBatchLoadTask";
const _CBLTR = "CreateBatchLoadTaskRequest";
const _CBLTRr = "CreateBatchLoadTaskResponse";
const _CC = "CsvConfiguration";
const _CD = "CreateDatabase";
const _CDR = "CreateDatabaseRequest";
const _CDRr = "CreateDatabaseResponse";
const _CE = "ConflictException";
const _CPIM = "CachePeriodInMinutes";
const _CPK = "CompositePartitionKey";
const _CRT = "ClientRequestToken";
const _CS = "ColumnSeparator";
const _CT = "CreationTime";
const _CTR = "CreateTableRequest";
const _CTRr = "CreateTableResponse";
const _CTl = "ClientToken";
const _CTr = "CreateTable";
const _D = "Database";
const _DBLT = "DescribeBatchLoadTask";
const _DBLTR = "DescribeBatchLoadTaskRequest";
const _DBLTRe = "DescribeBatchLoadTaskResponse";
const _DC = "DestinationColumn";
const _DD = "DeleteDatabase";
const _DDR = "DeleteDatabaseRequest";
const _DDRe = "DescribeDatabaseRequest";
const _DDRes = "DescribeDatabaseResponse";
const _DDe = "DescribeDatabase";
const _DE = "DescribeEndpoints";
const _DER = "DescribeEndpointsRequest";
const _DERe = "DescribeEndpointsResponse";
const _DF = "DataFormat";
const _DL = "DatabaseList";
const _DM = "DataModel";
const _DMC = "DataModelConfiguration";
const _DMSC = "DataModelS3Configuration";
const _DMi = "DimensionMappings";
const _DMim = "DimensionMapping";
const _DN = "DatabaseName";
const _DSC = "DataSourceConfiguration";
const _DSSC = "DataSourceS3Configuration";
const _DT = "DeleteTable";
const _DTR = "DeleteTableRequest";
const _DTRe = "DescribeTableRequest";
const _DTRes = "DescribeTableResponse";
const _DTe = "DescribeTable";
const _DVT = "DimensionValueType";
const _Da = "Databases";
const _Di = "Dimension";
const _Dim = "Dimensions";
const _E = "Endpoints";
const _EC = "EscapeChar";
const _EIR = "EnforcementInRecord";
const _EM = "ErrorMessage";
const _EMSW = "EnableMagneticStoreWrites";
const _EO = "EncryptionOption";
const _EV = "ExistingVersion";
const _En = "Endpoint";
const _FF = "FileFailures";
const _IEE = "InvalidEndpointException";
const _ISE = "InternalServerException";
const _K = "Key";
const _KKI = "KmsKeyId";
const _LBLT = "ListBatchLoadTasks";
const _LBLTR = "ListBatchLoadTasksRequest";
const _LBLTRi = "ListBatchLoadTasksResponse";
const _LD = "ListDatabases";
const _LDR = "ListDatabasesRequest";
const _LDRi = "ListDatabasesResponse";
const _LT = "ListTables";
const _LTFR = "ListTagsForResource";
const _LTFRR = "ListTagsForResourceRequest";
const _LTFRRi = "ListTagsForResourceResponse";
const _LTR = "ListTablesRequest";
const _LTRi = "ListTablesResponse";
const _LUT = "LastUpdatedTime";
const _M = "Message";
const _MMAM = "MultiMeasureAttributeMappings";
const _MMAML = "MultiMeasureAttributeMappingList";
const _MMAMu = "MultiMeasureAttributeMapping";
const _MMM = "MultiMeasureMappings";
const _MMML = "MixedMeasureMappingList";
const _MMMi = "MixedMeasureMappings";
const _MMMix = "MixedMeasureMapping";
const _MN = "MeasureName";
const _MNC = "MeasureNameColumn";
const _MR = "MaxResults";
const _MS = "MemoryStore";
const _MSRDL = "MagneticStoreRejectedDataLocation";
const _MSRPID = "MagneticStoreRetentionPeriodInDays";
const _MSRPIH = "MemoryStoreRetentionPeriodInHours";
const _MSWP = "MagneticStoreWriteProperties";
const _MSa = "MagneticStore";
const _MV = "MeasureValue";
const _MVT = "MeasureValueType";
const _MVe = "MeasureValues";
const _N = "Name";
const _NT = "NextToken";
const _NV = "NullValue";
const _OK = "ObjectKey";
const _OKP = "ObjectKeyPrefix";
const _PF = "ParseFailures";
const _PK = "PartitionKey";
const _PKL = "PartitionKeyList";
const _PR = "ProgressReport";
const _QC = "QuoteChar";
const _R = "Record";
const _RARN = "ResourceARN";
const _RBLT = "ResumeBatchLoadTask";
const _RBLTR = "ResumeBatchLoadTaskRequest";
const _RBLTRe = "ResumeBatchLoadTaskResponse";
const _RC = "ReportConfiguration";
const _RI = "RecordsIngested";
const _RIF = "RecordIngestionFailures";
const _RIe = "RecordIndex";
const _RNFE = "ResourceNotFoundException";
const _RP = "RecordsProcessed";
const _RPe = "RetentionProperties";
const _RR = "RejectedRecords";
const _RRE = "RejectedRecordsException";
const _RRe = "RejectedRecord";
const _RSC = "ReportS3Configuration";
const _RU = "ResumableUntil";
const _RV = "RecordVersion";
const _Re = "Reason";
const _Rec = "Records";
const _S = "Schema";
const _SC = "SourceColumn";
const _SCo = "S3Configuration";
const _SQEE = "ServiceQuotaExceededException";
const _T = "Tags";
const _TC = "TableCount";
const _TCi = "TimeColumn";
const _TDN = "TargetDatabaseName";
const _TE = "ThrottlingException";
const _TI = "TaskId";
const _TK = "TagKeys";
const _TL = "TableList";
const _TLa = "TagList";
const _TMMAN = "TargetMultiMeasureAttributeName";
const _TMMN = "TargetMultiMeasureName";
const _TMN = "TargetMeasureName";
const _TN = "TableName";
const _TR = "TagResource";
const _TRR = "TagResourceRequest";
const _TRRa = "TagResourceResponse";
const _TS = "TaskStatus";
const _TSa = "TableStatus";
const _TTN = "TargetTableName";
const _TU = "TimeUnit";
const _TWS = "TrimWhiteSpace";
const _Ta = "Table";
const _Tab = "Tables";
const _Tag = "Tag";
const _Ti = "Time";
const _To = "Total";
const _Ty = "Type";
const _UD = "UpdateDatabase";
const _UDR = "UpdateDatabaseRequest";
const _UDRp = "UpdateDatabaseResponse";
const _UR = "UntagResource";
const _URR = "UntagResourceRequest";
const _URRn = "UntagResourceResponse";
const _UT = "UpdateTable";
const _UTR = "UpdateTableRequest";
const _UTRp = "UpdateTableResponse";
const _V = "Value";
const _VE = "ValidationException";
const _Ve = "Version";
const _WR = "WriteRecords";
const _WRR = "WriteRecordsRequest";
const _WRRr = "WriteRecordsResponse";
const _c = "client";
const _e = "error";
const _hE = "httpError";
const _s = "smithy.ts.sdk.synthetic.com.amazonaws.timestreamwrite";
const _se = "server";
const n0 = "com.amazonaws.timestreamwrite";
import { TypeRegistry } from "@smithy/core/schema";
import { AccessDeniedException, ConflictException, InternalServerException, InvalidEndpointException, RejectedRecordsException, ResourceNotFoundException, ServiceQuotaExceededException, ThrottlingException, ValidationException, } from "../models/errors";
import { TimestreamWriteServiceException } from "../models/TimestreamWriteServiceException";
const _s_registry = TypeRegistry.for(_s);
export var TimestreamWriteServiceException$ = [-3, _s, "TimestreamWriteServiceException", 0, [], []];
_s_registry.registerError(TimestreamWriteServiceException$, TimestreamWriteServiceException);
const n0_registry = TypeRegistry.for(n0);
export var AccessDeniedException$ = [-3, n0, _ADE,
    { [_e]: _c, [_hE]: 403 },
    [_M],
    [0], 1
];
n0_registry.registerError(AccessDeniedException$, AccessDeniedException);
export var ConflictException$ = [-3, n0, _CE,
    { [_e]: _c, [_hE]: 409 },
    [_M],
    [0], 1
];
n0_registry.registerError(ConflictException$, ConflictException);
export var InternalServerException$ = [-3, n0, _ISE,
    { [_e]: _se, [_hE]: 500 },
    [_M],
    [0], 1
];
n0_registry.registerError(InternalServerException$, InternalServerException);
export var InvalidEndpointException$ = [-3, n0, _IEE,
    { [_e]: _c, [_hE]: 421 },
    [_M],
    [0]
];
n0_registry.registerError(InvalidEndpointException$, InvalidEndpointException);
export var RejectedRecordsException$ = [-3, n0, _RRE,
    { [_e]: _c, [_hE]: 419 },
    [_M, _RR],
    [0, () => RejectedRecords]
];
n0_registry.registerError(RejectedRecordsException$, RejectedRecordsException);
export var ResourceNotFoundException$ = [-3, n0, _RNFE,
    { [_e]: _c, [_hE]: 404 },
    [_M],
    [0]
];
n0_registry.registerError(ResourceNotFoundException$, ResourceNotFoundException);
export var ServiceQuotaExceededException$ = [-3, n0, _SQEE,
    { [_e]: _c, [_hE]: 402 },
    [_M],
    [0]
];
n0_registry.registerError(ServiceQuotaExceededException$, ServiceQuotaExceededException);
export var ThrottlingException$ = [-3, n0, _TE,
    { [_e]: _c, [_hE]: 429 },
    [_M],
    [0], 1
];
n0_registry.registerError(ThrottlingException$, ThrottlingException);
export var ValidationException$ = [-3, n0, _VE,
    { [_e]: _c, [_hE]: 400 },
    [_M],
    [0], 1
];
n0_registry.registerError(ValidationException$, ValidationException);
export const errorTypeRegistries = [
    _s_registry,
    n0_registry,
];
var ClientRequestToken = [0, n0, _CRT, 8, 0];
export var BatchLoadProgressReport$ = [3, n0, _BLPR,
    0,
    [_RP, _RI, _PF, _RIF, _FF, _BM],
    [1, 1, 1, 1, 1, 1]
];
export var BatchLoadTask$ = [3, n0, _BLT,
    0,
    [_TI, _TS, _DN, _TN, _CT, _LUT, _RU],
    [0, 0, 0, 0, 4, 4, 4]
];
export var BatchLoadTaskDescription$ = [3, n0, _BLTD,
    0,
    [_TI, _EM, _DSC, _PR, _RC, _DMC, _TDN, _TTN, _TS, _RV, _CT, _LUT, _RU],
    [0, 0, () => DataSourceConfiguration$, () => BatchLoadProgressReport$, () => ReportConfiguration$, () => DataModelConfiguration$, 0, 0, 0, 1, 4, 4, 4]
];
export var CreateBatchLoadTaskRequest$ = [3, n0, _CBLTR,
    0,
    [_DSC, _RC, _TDN, _TTN, _CTl, _DMC, _RV],
    [() => DataSourceConfiguration$, () => ReportConfiguration$, 0, 0, [() => ClientRequestToken, 4], () => DataModelConfiguration$, 1], 4
];
export var CreateBatchLoadTaskResponse$ = [3, n0, _CBLTRr,
    0,
    [_TI],
    [0], 1
];
export var CreateDatabaseRequest$ = [3, n0, _CDR,
    0,
    [_DN, _KKI, _T],
    [0, 0, () => TagList], 1
];
export var CreateDatabaseResponse$ = [3, n0, _CDRr,
    0,
    [_D],
    [() => Database$]
];
export var CreateTableRequest$ = [3, n0, _CTR,
    0,
    [_DN, _TN, _RPe, _T, _MSWP, _S],
    [0, 0, () => RetentionProperties$, () => TagList, () => MagneticStoreWriteProperties$, () => Schema$], 2
];
export var CreateTableResponse$ = [3, n0, _CTRr,
    0,
    [_Ta],
    [() => Table$]
];
export var CsvConfiguration$ = [3, n0, _CC,
    0,
    [_CS, _EC, _QC, _NV, _TWS],
    [0, 0, 0, 0, 2]
];
export var Database$ = [3, n0, _D,
    0,
    [_A, _DN, _TC, _KKI, _CT, _LUT],
    [0, 0, 1, 0, 4, 4]
];
export var DataModel$ = [3, n0, _DM,
    0,
    [_DMi, _TCi, _TU, _MMM, _MMMi, _MNC],
    [() => DimensionMappings, 0, 0, () => MultiMeasureMappings$, () => MixedMeasureMappingList, 0], 1
];
export var DataModelConfiguration$ = [3, n0, _DMC,
    0,
    [_DM, _DMSC],
    [() => DataModel$, () => DataModelS3Configuration$]
];
export var DataModelS3Configuration$ = [3, n0, _DMSC,
    0,
    [_BN, _OK],
    [0, 0]
];
export var DataSourceConfiguration$ = [3, n0, _DSC,
    0,
    [_DSSC, _DF, _CC],
    [() => DataSourceS3Configuration$, 0, () => CsvConfiguration$], 2
];
export var DataSourceS3Configuration$ = [3, n0, _DSSC,
    0,
    [_BN, _OKP],
    [0, 0], 1
];
export var DeleteDatabaseRequest$ = [3, n0, _DDR,
    0,
    [_DN],
    [0], 1
];
export var DeleteTableRequest$ = [3, n0, _DTR,
    0,
    [_DN, _TN],
    [0, 0], 2
];
export var DescribeBatchLoadTaskRequest$ = [3, n0, _DBLTR,
    0,
    [_TI],
    [0], 1
];
export var DescribeBatchLoadTaskResponse$ = [3, n0, _DBLTRe,
    0,
    [_BLTD],
    [() => BatchLoadTaskDescription$], 1
];
export var DescribeDatabaseRequest$ = [3, n0, _DDRe,
    0,
    [_DN],
    [0], 1
];
export var DescribeDatabaseResponse$ = [3, n0, _DDRes,
    0,
    [_D],
    [() => Database$]
];
export var DescribeEndpointsRequest$ = [3, n0, _DER,
    0,
    [],
    []
];
export var DescribeEndpointsResponse$ = [3, n0, _DERe,
    0,
    [_E],
    [() => Endpoints], 1
];
export var DescribeTableRequest$ = [3, n0, _DTRe,
    0,
    [_DN, _TN],
    [0, 0], 2
];
export var DescribeTableResponse$ = [3, n0, _DTRes,
    0,
    [_Ta],
    [() => Table$]
];
export var Dimension$ = [3, n0, _Di,
    0,
    [_N, _V, _DVT],
    [0, 0, 0], 2
];
export var DimensionMapping$ = [3, n0, _DMim,
    0,
    [_SC, _DC],
    [0, 0]
];
export var Endpoint$ = [3, n0, _En,
    0,
    [_Ad, _CPIM],
    [0, 1], 2
];
export var ListBatchLoadTasksRequest$ = [3, n0, _LBLTR,
    0,
    [_NT, _MR, _TS],
    [0, 1, 0]
];
export var ListBatchLoadTasksResponse$ = [3, n0, _LBLTRi,
    0,
    [_NT, _BLTa],
    [0, () => BatchLoadTaskList]
];
export var ListDatabasesRequest$ = [3, n0, _LDR,
    0,
    [_NT, _MR],
    [0, 1]
];
export var ListDatabasesResponse$ = [3, n0, _LDRi,
    0,
    [_Da, _NT],
    [() => DatabaseList, 0]
];
export var ListTablesRequest$ = [3, n0, _LTR,
    0,
    [_DN, _NT, _MR],
    [0, 0, 1]
];
export var ListTablesResponse$ = [3, n0, _LTRi,
    0,
    [_Tab, _NT],
    [() => TableList, 0]
];
export var ListTagsForResourceRequest$ = [3, n0, _LTFRR,
    0,
    [_RARN],
    [0], 1
];
export var ListTagsForResourceResponse$ = [3, n0, _LTFRRi,
    0,
    [_T],
    [() => TagList]
];
export var MagneticStoreRejectedDataLocation$ = [3, n0, _MSRDL,
    0,
    [_SCo],
    [() => S3Configuration$]
];
export var MagneticStoreWriteProperties$ = [3, n0, _MSWP,
    0,
    [_EMSW, _MSRDL],
    [2, () => MagneticStoreRejectedDataLocation$], 1
];
export var MeasureValue$ = [3, n0, _MV,
    0,
    [_N, _V, _Ty],
    [0, 0, 0], 3
];
export var MixedMeasureMapping$ = [3, n0, _MMMix,
    0,
    [_MVT, _MN, _SC, _TMN, _MMAM],
    [0, 0, 0, 0, () => MultiMeasureAttributeMappingList], 1
];
export var MultiMeasureAttributeMapping$ = [3, n0, _MMAMu,
    0,
    [_SC, _TMMAN, _MVT],
    [0, 0, 0], 1
];
export var MultiMeasureMappings$ = [3, n0, _MMM,
    0,
    [_MMAM, _TMMN],
    [() => MultiMeasureAttributeMappingList, 0], 1
];
export var PartitionKey$ = [3, n0, _PK,
    0,
    [_Ty, _N, _EIR],
    [0, 0, 0], 1
];
export var _Record$ = [3, n0, _R,
    0,
    [_Dim, _MN, _MV, _MVT, _Ti, _TU, _Ve, _MVe],
    [() => Dimensions, 0, 0, 0, 0, 0, 1, () => MeasureValues]
];
export var RecordsIngested$ = [3, n0, _RI,
    0,
    [_To, _MS, _MSa],
    [1, 1, 1]
];
export var RejectedRecord$ = [3, n0, _RRe,
    0,
    [_RIe, _Re, _EV],
    [1, 0, 1]
];
export var ReportConfiguration$ = [3, n0, _RC,
    0,
    [_RSC],
    [() => ReportS3Configuration$]
];
export var ReportS3Configuration$ = [3, n0, _RSC,
    0,
    [_BN, _OKP, _EO, _KKI],
    [0, 0, 0, 0], 1
];
export var ResumeBatchLoadTaskRequest$ = [3, n0, _RBLTR,
    0,
    [_TI],
    [0], 1
];
export var ResumeBatchLoadTaskResponse$ = [3, n0, _RBLTRe,
    0,
    [],
    []
];
export var RetentionProperties$ = [3, n0, _RPe,
    0,
    [_MSRPIH, _MSRPID],
    [1, 1], 2
];
export var S3Configuration$ = [3, n0, _SCo,
    0,
    [_BN, _OKP, _EO, _KKI],
    [0, 0, 0, 0]
];
export var Schema$ = [3, n0, _S,
    0,
    [_CPK],
    [() => PartitionKeyList]
];
export var Table$ = [3, n0, _Ta,
    0,
    [_A, _TN, _DN, _TSa, _RPe, _CT, _LUT, _MSWP, _S],
    [0, 0, 0, 0, () => RetentionProperties$, 4, 4, () => MagneticStoreWriteProperties$, () => Schema$]
];
export var Tag$ = [3, n0, _Tag,
    0,
    [_K, _V],
    [0, 0], 2
];
export var TagResourceRequest$ = [3, n0, _TRR,
    0,
    [_RARN, _T],
    [0, () => TagList], 2
];
export var TagResourceResponse$ = [3, n0, _TRRa,
    0,
    [],
    []
];
export var UntagResourceRequest$ = [3, n0, _URR,
    0,
    [_RARN, _TK],
    [0, 64 | 0], 2
];
export var UntagResourceResponse$ = [3, n0, _URRn,
    0,
    [],
    []
];
export var UpdateDatabaseRequest$ = [3, n0, _UDR,
    0,
    [_DN, _KKI],
    [0, 0], 2
];
export var UpdateDatabaseResponse$ = [3, n0, _UDRp,
    0,
    [_D],
    [() => Database$]
];
export var UpdateTableRequest$ = [3, n0, _UTR,
    0,
    [_DN, _TN, _RPe, _MSWP, _S],
    [0, 0, () => RetentionProperties$, () => MagneticStoreWriteProperties$, () => Schema$], 2
];
export var UpdateTableResponse$ = [3, n0, _UTRp,
    0,
    [_Ta],
    [() => Table$]
];
export var WriteRecordsRequest$ = [3, n0, _WRR,
    0,
    [_DN, _TN, _Rec, _CA],
    [0, 0, () => Records, () => _Record$], 3
];
export var WriteRecordsResponse$ = [3, n0, _WRRr,
    0,
    [_RI],
    [() => RecordsIngested$]
];
var __Unit = "unit";
var BatchLoadTaskList = [1, n0, _BLTL,
    0, () => BatchLoadTask$
];
var DatabaseList = [1, n0, _DL,
    0, () => Database$
];
var DimensionMappings = [1, n0, _DMi,
    0, () => DimensionMapping$
];
var Dimensions = [1, n0, _Dim,
    0, () => Dimension$
];
var Endpoints = [1, n0, _E,
    0, () => Endpoint$
];
var MeasureValues = [1, n0, _MVe,
    0, () => MeasureValue$
];
var MixedMeasureMappingList = [1, n0, _MMML,
    0, () => MixedMeasureMapping$
];
var MultiMeasureAttributeMappingList = [1, n0, _MMAML,
    0, () => MultiMeasureAttributeMapping$
];
var PartitionKeyList = [1, n0, _PKL,
    0, () => PartitionKey$
];
var Records = [1, n0, _Rec,
    0, () => _Record$
];
var RejectedRecords = [1, n0, _RR,
    0, () => RejectedRecord$
];
var TableList = [1, n0, _TL,
    0, () => Table$
];
var TagKeyList = 64 | 0;
var TagList = [1, n0, _TLa,
    0, () => Tag$
];
export var CreateBatchLoadTask$ = [9, n0, _CBLT,
    0, () => CreateBatchLoadTaskRequest$, () => CreateBatchLoadTaskResponse$
];
export var CreateDatabase$ = [9, n0, _CD,
    0, () => CreateDatabaseRequest$, () => CreateDatabaseResponse$
];
export var CreateTable$ = [9, n0, _CTr,
    0, () => CreateTableRequest$, () => CreateTableResponse$
];
export var DeleteDatabase$ = [9, n0, _DD,
    0, () => DeleteDatabaseRequest$, () => __Unit
];
export var DeleteTable$ = [9, n0, _DT,
    0, () => DeleteTableRequest$, () => __Unit
];
export var DescribeBatchLoadTask$ = [9, n0, _DBLT,
    0, () => DescribeBatchLoadTaskRequest$, () => DescribeBatchLoadTaskResponse$
];
export var DescribeDatabase$ = [9, n0, _DDe,
    0, () => DescribeDatabaseRequest$, () => DescribeDatabaseResponse$
];
export var DescribeEndpoints$ = [9, n0, _DE,
    0, () => DescribeEndpointsRequest$, () => DescribeEndpointsResponse$
];
export var DescribeTable$ = [9, n0, _DTe,
    0, () => DescribeTableRequest$, () => DescribeTableResponse$
];
export var ListBatchLoadTasks$ = [9, n0, _LBLT,
    0, () => ListBatchLoadTasksRequest$, () => ListBatchLoadTasksResponse$
];
export var ListDatabases$ = [9, n0, _LD,
    0, () => ListDatabasesRequest$, () => ListDatabasesResponse$
];
export var ListTables$ = [9, n0, _LT,
    0, () => ListTablesRequest$, () => ListTablesResponse$
];
export var ListTagsForResource$ = [9, n0, _LTFR,
    0, () => ListTagsForResourceRequest$, () => ListTagsForResourceResponse$
];
export var ResumeBatchLoadTask$ = [9, n0, _RBLT,
    0, () => ResumeBatchLoadTaskRequest$, () => ResumeBatchLoadTaskResponse$
];
export var TagResource$ = [9, n0, _TR,
    0, () => TagResourceRequest$, () => TagResourceResponse$
];
export var UntagResource$ = [9, n0, _UR,
    0, () => UntagResourceRequest$, () => UntagResourceResponse$
];
export var UpdateDatabase$ = [9, n0, _UD,
    0, () => UpdateDatabaseRequest$, () => UpdateDatabaseResponse$
];
export var UpdateTable$ = [9, n0, _UT,
    0, () => UpdateTableRequest$, () => UpdateTableResponse$
];
export var WriteRecords$ = [9, n0, _WR,
    0, () => WriteRecordsRequest$, () => WriteRecordsResponse$
];
