// @aws-sdk/client-timestream-write/dist-cjs/schemas/schemas_0.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MagneticStoreWriteProperties$ = exports.MagneticStoreRejectedDataLocation$ = exports.ListTagsForResourceResponse$ = exports.ListTagsForResourceRequest$ = exports.ListTablesResponse$ = exports.ListTablesRequest$ = exports.ListDatabasesResponse$ = exports.ListDatabasesRequest$ = exports.ListBatchLoadTasksResponse$ = exports.ListBatchLoadTasksRequest$ = exports.Endpoint$ = exports.DimensionMapping$ = exports.Dimension$ = exports.DescribeTableResponse$ = exports.DescribeTableRequest$ = exports.DescribeEndpointsResponse$ = exports.DescribeEndpointsRequest$ = exports.DescribeDatabaseResponse$ = exports.DescribeDatabaseRequest$ = exports.DescribeBatchLoadTaskResponse$ = exports.DescribeBatchLoadTaskRequest$ = exports.DeleteTableRequest$ = exports.DeleteDatabaseRequest$ = exports.DataSourceS3Configuration$ = exports.DataSourceConfiguration$ = exports.DataModelS3Configuration$ = exports.DataModelConfiguration$ = exports.DataModel$ = exports.Database$ = exports.CsvConfiguration$ = exports.CreateTableResponse$ = exports.CreateTableRequest$ = exports.CreateDatabaseResponse$ = exports.CreateDatabaseRequest$ = exports.CreateBatchLoadTaskResponse$ = exports.CreateBatchLoadTaskRequest$ = exports.BatchLoadTaskDescription$ = exports.BatchLoadTask$ = exports.BatchLoadProgressReport$ = exports.errorTypeRegistries = exports.ValidationException$ = exports.ThrottlingException$ = exports.ServiceQuotaExceededException$ = exports.ResourceNotFoundException$ = exports.RejectedRecordsException$ = exports.InvalidEndpointException$ = exports.InternalServerException$ = exports.ConflictException$ = exports.AccessDeniedException$ = exports.TimestreamWriteServiceException$ = void 0;
exports.WriteRecords$ = exports.UpdateTable$ = exports.UpdateDatabase$ = exports.UntagResource$ = exports.TagResource$ = exports.ResumeBatchLoadTask$ = exports.ListTagsForResource$ = exports.ListTables$ = exports.ListDatabases$ = exports.ListBatchLoadTasks$ = exports.DescribeTable$ = exports.DescribeEndpoints$ = exports.DescribeDatabase$ = exports.DescribeBatchLoadTask$ = exports.DeleteTable$ = exports.DeleteDatabase$ = exports.CreateTable$ = exports.CreateDatabase$ = exports.CreateBatchLoadTask$ = exports.WriteRecordsResponse$ = exports.WriteRecordsRequest$ = exports.UpdateTableResponse$ = exports.UpdateTableRequest$ = exports.UpdateDatabaseResponse$ = exports.UpdateDatabaseRequest$ = exports.UntagResourceResponse$ = exports.UntagResourceRequest$ = exports.TagResourceResponse$ = exports.TagResourceRequest$ = exports.Tag$ = exports.Table$ = exports.Schema$ = exports.S3Configuration$ = exports.RetentionProperties$ = exports.ResumeBatchLoadTaskResponse$ = exports.ResumeBatchLoadTaskRequest$ = exports.ReportS3Configuration$ = exports.ReportConfiguration$ = exports.RejectedRecord$ = exports.RecordsIngested$ = exports._Record$ = exports.PartitionKey$ = exports.MultiMeasureMappings$ = exports.MultiMeasureAttributeMapping$ = exports.MixedMeasureMapping$ = exports.MeasureValue$ = void 0;
const _A = "Arn";
const _ADE = "AccessDeniedException";
const _Ad = "Address";
const _BLPR = "BatchLoadProgressReport";
const _BLT = "BatchLoadTask";
const _BLTD = "BatchLoadTaskDescription";
const _BLTL = "BatchLoadTaskList";
const _BLTa = "BatchLoadTasks";
const _BM = "BytesMetered";
const _BN = "BucketName";
const _CA = "CommonAttributes";
const _CBLT = "CreateBatchLoadTask";
const _CBLTR = "CreateBatchLoadTaskRequest";
const _CBLTRr = "CreateBatchLoadTaskResponse";
const _CC = "CsvConfiguration";
const _CD = "CreateDatabase";
const _CDR = "CreateDatabaseRequest";
const _CDRr = "CreateDatabaseResponse";
const _CE = "ConflictException";
const _CPIM = "CachePeriodInMinutes";
const _CPK = "CompositePartitionKey";
const _CRT = "ClientRequestToken";
const _CS = "ColumnSeparator";
const _CT = "CreationTime";
const _CTR = "CreateTableRequest";
const _CTRr = "CreateTableResponse";
const _CTl = "ClientToken";
const _CTr = "CreateTable";
const _D = "Database";
const _DBLT = "DescribeBatchLoadTask";
const _DBLTR = "DescribeBatchLoadTaskRequest";
const _DBLTRe = "DescribeBatchLoadTaskResponse";
const _DC = "DestinationColumn";
const _DD = "DeleteDatabase";
const _DDR = "DeleteDatabaseRequest";
const _DDRe = "DescribeDatabaseRequest";
const _DDRes = "DescribeDatabaseResponse";
const _DDe = "DescribeDatabase";
const _DE = "DescribeEndpoints";
const _DER = "DescribeEndpointsRequest";
const _DERe = "DescribeEndpointsResponse";
const _DF = "DataFormat";
const _DL = "DatabaseList";
const _DM = "DataModel";
const _DMC = "DataModelConfiguration";
const _DMSC = "DataModelS3Configuration";
const _DMi = "DimensionMappings";
const _DMim = "DimensionMapping";
const _DN = "DatabaseName";
const _DSC = "DataSourceConfiguration";
const _DSSC = "DataSourceS3Configuration";
const _DT = "DeleteTable";
const _DTR = "DeleteTableRequest";
const _DTRe = "DescribeTableRequest";
const _DTRes = "DescribeTableResponse";
const _DTe = "DescribeTable";
const _DVT = "DimensionValueType";
const _Da = "Databases";
const _Di = "Dimension";
const _Dim = "Dimensions";
const _E = "Endpoints";
const _EC = "EscapeChar";
const _EIR = "EnforcementInRecord";
const _EM = "ErrorMessage";
const _EMSW = "EnableMagneticStoreWrites";
const _EO = "EncryptionOption";
const _EV = "ExistingVersion";
const _En = "Endpoint";
const _FF = "FileFailures";
const _IEE = "InvalidEndpointException";
const _ISE = "InternalServerException";
const _K = "Key";
const _KKI = "KmsKeyId";
const _LBLT = "ListBatchLoadTasks";
const _LBLTR = "ListBatchLoadTasksRequest";
const _LBLTRi = "ListBatchLoadTasksResponse";
const _LD = "ListDatabases";
const _LDR = "ListDatabasesRequest";
const _LDRi = "ListDatabasesResponse";
const _LT = "ListTables";
const _LTFR = "ListTagsForResource";
const _LTFRR = "ListTagsForResourceRequest";
const _LTFRRi = "ListTagsForResourceResponse";
const _LTR = "ListTablesRequest";
const _LTRi = "ListTablesResponse";
const _LUT = "LastUpdatedTime";
const _M = "Message";
const _MMAM = "MultiMeasureAttributeMappings";
const _MMAML = "MultiMeasureAttributeMappingList";
const _MMAMu = "MultiMeasureAttributeMapping";
const _MMM = "MultiMeasureMappings";
const _MMML = "MixedMeasureMappingList";
const _MMMi = "MixedMeasureMappings";
const _MMMix = "MixedMeasureMapping";
const _MN = "MeasureName";
const _MNC = "MeasureNameColumn";
const _MR = "MaxResults";
const _MS = "MemoryStore";
const _MSRDL = "MagneticStoreRejectedDataLocation";
const _MSRPID = "MagneticStoreRetentionPeriodInDays";
const _MSRPIH = "MemoryStoreRetentionPeriodInHours";
const _MSWP = "MagneticStoreWriteProperties";
const _MSa = "MagneticStore";
const _MV = "MeasureValue";
const _MVT = "MeasureValueType";
const _MVe = "MeasureValues";
const _N = "Name";
const _NT = "NextToken";
const _NV = "NullValue";
const _OK = "ObjectKey";
const _OKP = "ObjectKeyPrefix";
const _PF = "ParseFailures";
const _PK = "PartitionKey";
const _PKL = "PartitionKeyList";
const _PR = "ProgressReport";
const _QC = "QuoteChar";
const _R = "Record";
const _RARN = "ResourceARN";
const _RBLT = "ResumeBatchLoadTask";
const _RBLTR = "ResumeBatchLoadTaskRequest";
const _RBLTRe = "ResumeBatchLoadTaskResponse";
const _RC = "ReportConfiguration";
const _RI = "RecordsIngested";
const _RIF = "RecordIngestionFailures";
const _RIe = "RecordIndex";
const _RNFE = "ResourceNotFoundException";
const _RP = "RecordsProcessed";
const _RPe = "RetentionProperties";
const _RR = "RejectedRecords";
const _RRE = "RejectedRecordsException";
const _RRe = "RejectedRecord";
const _RSC = "ReportS3Configuration";
const _RU = "ResumableUntil";
const _RV = "RecordVersion";
const _Re = "Reason";
const _Rec = "Records";
const _S = "Schema";
const _SC = "SourceColumn";
const _SCo = "S3Configuration";
const _SQEE = "ServiceQuotaExceededException";
const _T = "Tags";
const _TC = "TableCount";
const _TCi = "TimeColumn";
const _TDN = "TargetDatabaseName";
const _TE = "ThrottlingException";
const _TI = "TaskId";
const _TK = "TagKeys";
const _TL = "TableList";
const _TLa = "TagList";
const _TMMAN = "TargetMultiMeasureAttributeName";
const _TMMN = "TargetMultiMeasureName";
const _TMN = "TargetMeasureName";
const _TN = "TableName";
const _TR = "TagResource";
const _TRR = "TagResourceRequest";
const _TRRa = "TagResourceResponse";
const _TS = "TaskStatus";
const _TSa = "TableStatus";
const _TTN = "TargetTableName";
const _TU = "TimeUnit";
const _TWS = "TrimWhiteSpace";
const _Ta = "Table";
const _Tab = "Tables";
const _Tag = "Tag";
const _Ti = "Time";
const _To = "Total";
const _Ty = "Type";
const _UD = "UpdateDatabase";
const _UDR = "UpdateDatabaseRequest";
const _UDRp = "UpdateDatabaseResponse";
const _UR = "UntagResource";
const _URR = "UntagResourceRequest";
const _URRn = "UntagResourceResponse";
const _UT = "UpdateTable";
const _UTR = "UpdateTableRequest";
const _UTRp = "UpdateTableResponse";
const _V = "Value";
const _VE = "ValidationException";
const _Ve = "Version";
const _WR = "WriteRecords";
const _WRR = "WriteRecordsRequest";
const _WRRr = "WriteRecordsResponse";
const _c = "client";
const _e = "error";
const _hE = "httpError";
const _s = "smithy.ts.sdk.synthetic.com.amazonaws.timestreamwrite";
const _se = "server";
const n0 = "com.amazonaws.timestreamwrite";
const schema_1 = require("@smithy/core/schema");
const errors_1 = require("../models/errors");
const TimestreamWriteServiceException_1 = require("../models/TimestreamWriteServiceException");
const _s_registry = schema_1.TypeRegistry.for(_s);
exports.TimestreamWriteServiceException$ = [-3, _s, "TimestreamWriteServiceException", 0, [], []];
_s_registry.registerError(exports.TimestreamWriteServiceException$, TimestreamWriteServiceException_1.TimestreamWriteServiceException);
const n0_registry = schema_1.TypeRegistry.for(n0);
exports.AccessDeniedException$ = [-3, n0, _ADE,
    { [_e]: _c, [_hE]: 403 },
    [_M],
    [0], 1
];
n0_registry.registerError(exports.AccessDeniedException$, errors_1.AccessDeniedException);
exports.ConflictException$ = [-3, n0, _CE,
    { [_e]: _c, [_hE]: 409 },
    [_M],
    [0], 1
];
n0_registry.registerError(exports.ConflictException$, errors_1.ConflictException);
exports.InternalServerException$ = [-3, n0, _ISE,
    { [_e]: _se, [_hE]: 500 },
    [_M],
    [0], 1
];
n0_registry.registerError(exports.InternalServerException$, errors_1.InternalServerException);
exports.InvalidEndpointException$ = [-3, n0, _IEE,
    { [_e]: _c, [_hE]: 421 },
    [_M],
    [0]
];
n0_registry.registerError(exports.InvalidEndpointException$, errors_1.InvalidEndpointException);
exports.RejectedRecordsException$ = [-3, n0, _RRE,
    { [_e]: _c, [_hE]: 419 },
    [_M, _RR],
    [0, () => RejectedRecords]
];
n0_registry.registerError(exports.RejectedRecordsException$, errors_1.RejectedRecordsException);
exports.ResourceNotFoundException$ = [-3, n0, _RNFE,
    { [_e]: _c, [_hE]: 404 },
    [_M],
    [0]
];
n0_registry.registerError(exports.ResourceNotFoundException$, errors_1.ResourceNotFoundException);
exports.ServiceQuotaExceededException$ = [-3, n0, _SQEE,
    { [_e]: _c, [_hE]: 402 },
    [_M],
    [0]
];
n0_registry.registerError(exports.ServiceQuotaExceededException$, errors_1.ServiceQuotaExceededException);
exports.ThrottlingException$ = [-3, n0, _TE,
    { [_e]: _c, [_hE]: 429 },
    [_M],
    [0], 1
];
n0_registry.registerError(exports.ThrottlingException$, errors_1.ThrottlingException);
exports.ValidationException$ = [-3, n0, _VE,
    { [_e]: _c, [_hE]: 400 },
    [_M],
    [0], 1
];
n0_registry.registerError(exports.ValidationException$, errors_1.ValidationException);
exports.errorTypeRegistries = [
    _s_registry,
    n0_registry,
];
var ClientRequestToken = [0, n0, _CRT, 8, 0];
exports.BatchLoadProgressReport$ = [3, n0, _BLPR,
    0,
    [_RP, _RI, _PF, _RIF, _FF, _BM],
    [1, 1, 1, 1, 1, 1]
];
exports.BatchLoadTask$ = [3, n0, _BLT,
    0,
    [_TI, _TS, _DN, _TN, _CT, _LUT, _RU],
    [0, 0, 0, 0, 4, 4, 4]
];
exports.BatchLoadTaskDescription$ = [3, n0, _BLTD,
    0,
    [_TI, _EM, _DSC, _PR, _RC, _DMC, _TDN, _TTN, _TS, _RV, _CT, _LUT, _RU],
    [0, 0, () => exports.DataSourceConfiguration$, () => exports.BatchLoadProgressReport$, () => exports.ReportConfiguration$, () => exports.DataModelConfiguration$, 0, 0, 0, 1, 4, 4, 4]
];
exports.CreateBatchLoadTaskRequest$ = [3, n0, _CBLTR,
    0,
    [_DSC, _RC, _TDN, _TTN, _CTl, _DMC, _RV],
    [() => exports.DataSourceConfiguration$, () => exports.ReportConfiguration$, 0, 0, [() => ClientRequestToken, 4], () => exports.DataModelConfiguration$, 1], 4
];
exports.CreateBatchLoadTaskResponse$ = [3, n0, _CBLTRr,
    0,
    [_TI],
    [0], 1
];
exports.CreateDatabaseRequest$ = [3, n0, _CDR,
    0,
    [_DN, _KKI, _T],
    [0, 0, () => TagList], 1
];
exports.CreateDatabaseResponse$ = [3, n0, _CDRr,
    0,
    [_D],
    [() => exports.Database$]
];
exports.CreateTableRequest$ = [3, n0, _CTR,
    0,
    [_DN, _TN, _RPe, _T, _MSWP, _S],
    [0, 0, () => exports.RetentionProperties$, () => TagList, () => exports.MagneticStoreWriteProperties$, () => exports.Schema$], 2
];
exports.CreateTableResponse$ = [3, n0, _CTRr,
    0,
    [_Ta],
    [() => exports.Table$]
];
exports.CsvConfiguration$ = [3, n0, _CC,
    0,
    [_CS, _EC, _QC, _NV, _TWS],
    [0, 0, 0, 0, 2]
];
exports.Database$ = [3, n0, _D,
    0,
    [_A, _DN, _TC, _KKI, _CT, _LUT],
    [0, 0, 1, 0, 4, 4]
];
exports.DataModel$ = [3, n0, _DM,
    0,
    [_DMi, _TCi, _TU, _MMM, _MMMi, _MNC],
    [() => DimensionMappings, 0, 0, () => exports.MultiMeasureMappings$, () => MixedMeasureMappingList, 0], 1
];
exports.DataModelConfiguration$ = [3, n0, _DMC,
    0,
    [_DM, _DMSC],
    [() => exports.DataModel$, () => exports.DataModelS3Configuration$]
];
exports.DataModelS3Configuration$ = [3, n0, _DMSC,
    0,
    [_BN, _OK],
    [0, 0]
];
exports.DataSourceConfiguration$ = [3, n0, _DSC,
    0,
    [_DSSC, _DF, _CC],
    [() => exports.DataSourceS3Configuration$, 0, () => exports.CsvConfiguration$], 2
];
exports.DataSourceS3Configuration$ = [3, n0, _DSSC,
    0,
    [_BN, _OKP],
    [0, 0], 1
];
exports.DeleteDatabaseRequest$ = [3, n0, _DDR,
    0,
    [_DN],
    [0], 1
];
exports.DeleteTableRequest$ = [3, n0, _DTR,
    0,
    [_DN, _TN],
    [0, 0], 2
];
exports.DescribeBatchLoadTaskRequest$ = [3, n0, _DBLTR,
    0,
    [_TI],
    [0], 1
];
exports.DescribeBatchLoadTaskResponse$ = [3, n0, _DBLTRe,
    0,
    [_BLTD],
    [() => exports.BatchLoadTaskDescription$], 1
];
exports.DescribeDatabaseRequest$ = [3, n0, _DDRe,
    0,
    [_DN],
    [0], 1
];
exports.DescribeDatabaseResponse$ = [3, n0, _DDRes,
    0,
    [_D],
    [() => exports.Database$]
];
exports.DescribeEndpointsRequest$ = [3, n0, _DER,
    0,
    [],
    []
];
exports.DescribeEndpointsResponse$ = [3, n0, _DERe,
    0,
    [_E],
    [() => Endpoints], 1
];
exports.DescribeTableRequest$ = [3, n0, _DTRe,
    0,
    [_DN, _TN],
    [0, 0], 2
];
exports.DescribeTableResponse$ = [3, n0, _DTRes,
    0,
    [_Ta],
    [() => exports.Table$]
];
exports.Dimension$ = [3, n0, _Di,
    0,
    [_N, _V, _DVT],
    [0, 0, 0], 2
];
exports.DimensionMapping$ = [3, n0, _DMim,
    0,
    [_SC, _DC],
    [0, 0]
];
exports.Endpoint$ = [3, n0, _En,
    0,
    [_Ad, _CPIM],
    [0, 1], 2
];
exports.ListBatchLoadTasksRequest$ = [3, n0, _LBLTR,
    0,
    [_NT, _MR, _TS],
    [0, 1, 0]
];
exports.ListBatchLoadTasksResponse$ = [3, n0, _LBLTRi,
    0,
    [_NT, _BLTa],
    [0, () => BatchLoadTaskList]
];
exports.ListDatabasesRequest$ = [3, n0, _LDR,
    0,
    [_NT, _MR],
    [0, 1]
];
exports.ListDatabasesResponse$ = [3, n0, _LDRi,
    0,
    [_Da, _NT],
    [() => DatabaseList, 0]
];
exports.ListTablesRequest$ = [3, n0, _LTR,
    0,
    [_DN, _NT, _MR],
    [0, 0, 1]
];
exports.ListTablesResponse$ = [3, n0, _LTRi,
    0,
    [_Tab, _NT],
    [() => TableList, 0]
];
exports.ListTagsForResourceRequest$ = [3, n0, _LTFRR,
    0,
    [_RARN],
    [0], 1
];
exports.ListTagsForResourceResponse$ = [3, n0, _LTFRRi,
    0,
    [_T],
    [() => TagList]
];
exports.MagneticStoreRejectedDataLocation$ = [3, n0, _MSRDL,
    0,
    [_SCo],
    [() => exports.S3Configuration$]
];
exports.MagneticStoreWriteProperties$ = [3, n0, _MSWP,
    0,
    [_EMSW, _MSRDL],
    [2, () => exports.MagneticStoreRejectedDataLocation$], 1
];
exports.MeasureValue$ = [3, n0, _MV,
    0,
    [_N, _V, _Ty],
    [0, 0, 0], 3
];
exports.MixedMeasureMapping$ = [3, n0, _MMMix,
    0,
    [_MVT, _MN, _SC, _TMN, _MMAM],
    [0, 0, 0, 0, () => MultiMeasureAttributeMappingList], 1
];
exports.MultiMeasureAttributeMapping$ = [3, n0, _MMAMu,
    0,
    [_SC, _TMMAN, _MVT],
    [0, 0, 0], 1
];
exports.MultiMeasureMappings$ = [3, n0, _MMM,
    0,
    [_MMAM, _TMMN],
    [() => MultiMeasureAttributeMappingList, 0], 1
];
exports.PartitionKey$ = [3, n0, _PK,
    0,
    [_Ty, _N, _EIR],
    [0, 0, 0], 1
];
exports._Record$ = [3, n0, _R,
    0,
    [_Dim, _MN, _MV, _MVT, _Ti, _TU, _Ve, _MVe],
    [() => Dimensions, 0, 0, 0, 0, 0, 1, () => MeasureValues]
];
exports.RecordsIngested$ = [3, n0, _RI,
    0,
    [_To, _MS, _MSa],
    [1, 1, 1]
];
exports.RejectedRecord$ = [3, n0, _RRe,
    0,
    [_RIe, _Re, _EV],
    [1, 0, 1]
];
exports.ReportConfiguration$ = [3, n0, _RC,
    0,
    [_RSC],
    [() => exports.ReportS3Configuration$]
];
exports.ReportS3Configuration$ = [3, n0, _RSC,
    0,
    [_BN, _OKP, _EO, _KKI],
    [0, 0, 0, 0], 1
];
exports.ResumeBatchLoadTaskRequest$ = [3, n0, _RBLTR,
    0,
    [_TI],
    [0], 1
];
exports.ResumeBatchLoadTaskResponse$ = [3, n0, _RBLTRe,
    0,
    [],
    []
];
exports.RetentionProperties$ = [3, n0, _RPe,
    0,
    [_MSRPIH, _MSRPID],
    [1, 1], 2
];
exports.S3Configuration$ = [3, n0, _SCo,
    0,
    [_BN, _OKP, _EO, _KKI],
    [0, 0, 0, 0]
];
exports.Schema$ = [3, n0, _S,
    0,
    [_CPK],
    [() => PartitionKeyList]
];
exports.Table$ = [3, n0, _Ta,
    0,
    [_A, _TN, _DN, _TSa, _RPe, _CT, _LUT, _MSWP, _S],
    [0, 0, 0, 0, () => exports.RetentionProperties$, 4, 4, () => exports.MagneticStoreWriteProperties$, () => exports.Schema$]
];
exports.Tag$ = [3, n0, _Tag,
    0,
    [_K, _V],
    [0, 0], 2
];
exports.TagResourceRequest$ = [3, n0, _TRR,
    0,
    [_RARN, _T],
    [0, () => TagList], 2
];
exports.TagResourceResponse$ = [3, n0, _TRRa,
    0,
    [],
    []
];
exports.UntagResourceRequest$ = [3, n0, _URR,
    0,
    [_RARN, _TK],
    [0, 64 | 0], 2
];
exports.UntagResourceResponse$ = [3, n0, _URRn,
    0,
    [],
    []
];
exports.UpdateDatabaseRequest$ = [3, n0, _UDR,
    0,
    [_DN, _KKI],
    [0, 0], 2
];
exports.UpdateDatabaseResponse$ = [3, n0, _UDRp,
    0,
    [_D],
    [() => exports.Database$]
];
exports.UpdateTableRequest$ = [3, n0, _UTR,
    0,
    [_DN, _TN, _RPe, _MSWP, _S],
    [0, 0, () => exports.RetentionProperties$, () => exports.MagneticStoreWriteProperties$, () => exports.Schema$], 2
];
exports.UpdateTableResponse$ = [3, n0, _UTRp,
    0,
    [_Ta],
    [() => exports.Table$]
];
exports.WriteRecordsRequest$ = [3, n0, _WRR,
    0,
    [_DN, _TN, _Rec, _CA],
    [0, 0, () => Records, () => exports._Record$], 3
];
exports.WriteRecordsResponse$ = [3, n0, _WRRr,
    0,
    [_RI],
    [() => exports.RecordsIngested$]
];
var __Unit = "unit";
var BatchLoadTaskList = [1, n0, _BLTL,
    0, () => exports.BatchLoadTask$
];
var DatabaseList = [1, n0, _DL,
    0, () => exports.Database$
];
var DimensionMappings = [1, n0, _DMi,
    0, () => exports.DimensionMapping$
];
var Dimensions = [1, n0, _Dim,
    0, () => exports.Dimension$
];
var Endpoints = [1, n0, _E,
    0, () => exports.Endpoint$
];
var MeasureValues = [1, n0, _MVe,
    0, () => exports.MeasureValue$
];
var MixedMeasureMappingList = [1, n0, _MMML,
    0, () => exports.MixedMeasureMapping$
];
var MultiMeasureAttributeMappingList = [1, n0, _MMAML,
    0, () => exports.MultiMeasureAttributeMapping$
];
var PartitionKeyList = [1, n0, _PKL,
    0, () => exports.PartitionKey$
];
var Records = [1, n0, _Rec,
    0, () => exports._Record$
];
var RejectedRecords = [1, n0, _RR,
    0, () => exports.RejectedRecord$
];
var TableList = [1, n0, _TL,
    0, () => exports.Table$
];
var TagKeyList = 64 | 0;
var TagList = [1, n0, _TLa,
    0, () => exports.Tag$
];
exports.CreateBatchLoadTask$ = [9, n0, _CBLT,
    0, () => exports.CreateBatchLoadTaskRequest$, () => exports.CreateBatchLoadTaskResponse$
];
exports.CreateDatabase$ = [9, n0, _CD,
    0, () => exports.CreateDatabaseRequest$, () => exports.CreateDatabaseResponse$
];
exports.CreateTable$ = [9, n0, _CTr,
    0, () => exports.CreateTableRequest$, () => exports.CreateTableResponse$
];
exports.DeleteDatabase$ = [9, n0, _DD,
    0, () => exports.DeleteDatabaseRequest$, () => __Unit
];
exports.DeleteTable$ = [9, n0, _DT,
    0, () => exports.DeleteTableRequest$, () => __Unit
];
exports.DescribeBatchLoadTask$ = [9, n0, _DBLT,
    0, () => exports.DescribeBatchLoadTaskRequest$, () => exports.DescribeBatchLoadTaskResponse$
];
exports.DescribeDatabase$ = [9, n0, _DDe,
    0, () => exports.DescribeDatabaseRequest$, () => exports.DescribeDatabaseResponse$
];
exports.DescribeEndpoints$ = [9, n0, _DE,
    0, () => exports.DescribeEndpointsRequest$, () => exports.DescribeEndpointsResponse$
];
exports.DescribeTable$ = [9, n0, _DTe,
    0, () => exports.DescribeTableRequest$, () => exports.DescribeTableResponse$
];
exports.ListBatchLoadTasks$ = [9, n0, _LBLT,
    0, () => exports.ListBatchLoadTasksRequest$, () => exports.ListBatchLoadTasksResponse$
];
exports.ListDatabases$ = [9, n0, _LD,
    0, () => exports.ListDatabasesRequest$, () => exports.ListDatabasesResponse$
];
exports.ListTables$ = [9, n0, _LT,
    0, () => exports.ListTablesRequest$, () => exports.ListTablesResponse$
];
exports.ListTagsForResource$ = [9, n0, _LTFR,
    0, () => exports.ListTagsForResourceRequest$, () => exports.ListTagsForResourceResponse$
];
exports.ResumeBatchLoadTask$ = [9, n0, _RBLT,
    0, () => exports.ResumeBatchLoadTaskRequest$, () => exports.ResumeBatchLoadTaskResponse$
];
exports.TagResource$ = [9, n0, _TR,
    0, () => exports.TagResourceRequest$, () => exports.TagResourceResponse$
];
exports.UntagResource$ = [9, n0, _UR,
    0, () => exports.UntagResourceRequest$, () => exports.UntagResourceResponse$
];
exports.UpdateDatabase$ = [9, n0, _UD,
    0, () => exports.UpdateDatabaseRequest$, () => exports.UpdateDatabaseResponse$
];
exports.UpdateTable$ = [9, n0, _UT,
    0, () => exports.UpdateTableRequest$, () => exports.UpdateTableResponse$
];
exports.WriteRecords$ = [9, n0, _WR,
    0, () => exports.WriteRecordsRequest$, () => exports.WriteRecordsResponse$
];
