// @aws-sdk/client-timestream-write/dist-es/models/enums.js
export const BatchLoadDataFormat = {
    CSV: "CSV",
};
export const BatchLoadStatus = {
    CREATED: "CREATED",
    FAILED: "FAILED",
    IN_PROGRESS: "IN_PROGRESS",
    PENDING_RESUME: "PENDING_RESUME",
    PROGRESS_STOPPED: "PROGRESS_STOPPED",
    SUCCEEDED: "SUCCEEDED",
};
export const MeasureValueType = {
    BIGINT: "BIGINT",
    BOOLEAN: "BOOLEAN",
    DOUBLE: "DOUBLE",
    MULTI: "MULTI",
    TIMESTAMP: "TIMESTAMP",
    VARCHAR: "VARCHAR",
};
export const ScalarMeasureValueType = {
    BIGINT: "BIGINT",
    BOOLEAN: "BOOLEAN",
    DOUBLE: "DOUBLE",
    TIMESTAMP: "TIMESTAMP",
    VARCHAR: "VARCHAR",
};
export const TimeUnit = {
    MICROSECONDS: "MICROSECONDS",
    MILLISECONDS: "MILLISECONDS",
    NANOSECONDS: "NANOSECONDS",
    SECONDS: "SECONDS",
};
export const S3EncryptionOption = {
    SSE_KMS: "SSE_KMS",
    SSE_S3: "SSE_S3",
};
export const PartitionKeyEnforcementLevel = {
    OPTIONAL: "OPTIONAL",
    REQUIRED: "REQUIRED",
};
export const PartitionKeyType = {
    DIMENSION: "DIMENSION",
    MEASURE: "MEASURE",
};
export const TableStatus = {
    ACTIVE: "ACTIVE",
    DELETING: "DELETING",
    RESTORING: "RESTORING",
};
export const DimensionValueType = {
    VARCHAR: "VARCHAR",
};
